x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -732,11 +748,19 @@ public void messageReceived(DataNodeRequest request, TransportChannel channel, T
             final ActionListener<ComputeResponse> listener = new ChannelActionListener<>(channel);
             final ExchangeSinkExec reducePlan;
             if (request.plan() instanceof ExchangeSinkExec plan) {
+                var fragments = plan.collectFirstChildren(FragmentExec.class::isInstance);
+                if (fragments.isEmpty()) {
+                    listener.onFailure(new IllegalStateException("expected a fragment plan for a remote compute; got " + request.plan()));
+                    return;
+                }
+
+                var localExchangeSource = new ExchangeSourceExec(plan.source(), plan.output(), plan.isIntermediateAgg());
+                FragmentExec fragment = (FragmentExec) fragments.get(0);
                 reducePlan = new ExchangeSinkExec(
                     plan.source(),
                     plan.output(),
                     plan.isIntermediateAgg(),
-                    new ExchangeSourceExec(plan.source(), plan.output(), plan.isIntermediateAgg())
+                    fragment.reducer() != null ? fragment.reducer().replaceChildren(List.of(localExchangeSource)) : localExchangeSource
```

Comment:
Use the physicalplan reducer node that we got from the coordinator to link it in the sink-exchange plan, to become as sink-reducer-source.